infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -117,15 +118,15 @@ public StreamLogFiles(ServerContext serverContext, boolean noVerify) {
         channelsToSync = new HashSet<>();
         this.verify = !noVerify;
         this.dataStore = StreamLogDataStore.builder().dataStore(serverContext.getDataStore()).build();
-        String logSizeQuotaParam = (String) serverContext.getServerConfig().get("--log-size-quota-bytes");
+        String logSizeLimitPercParam = (String) serverContext.getServerConfig().get("--log-size-quota-percentage");
 
-        logSizeLimit = Long.parseLong(logSizeQuotaParam);
+        logSizeLimitPercentage = Double.parseDouble(logSizeLimitPercParam);
 
         initStreamLogDirectory();
 
         long initialLogSize = estimateSize(logDir);
         log.info("StreamLogFiles: {} size is {} bytes, limit {}", logDir, initialLogSize, logSizeLimit);
-        logSizeQuota = new ResourceQuota("LogSizeQuota", logSizeLimit == -1 ? Long.MAX_VALUE : logSizeLimit);
+        logSizeQuota = new ResourceQuota("LogSizeQuota", logSizeLimit);
```

Comment:
Is there a way to explicitly initialize `logSizeLimit` in this constructor? I see the initialization of `logSizeLimit` is inside `initStreamLogDirectory()`, which seems confusing, right?

Reply:
If we move it to the constructor we have to duplicate the IOException handling logic again, which might be uglier right?